LiteCore/Database/DatabaseImpl.cc

```diff
@@ -576,12 +576,11 @@ namespace litecore {
                 DebugAssert(isValidScopeNameOrDefault(scope));
                 name.setStart(slash + 1);
             }
-            DebugAssert((name == kC4DefaultCollectionName && scope == kC4DefaultScopeID)
```

Comment:
So, we have have an instance of violating this assertion. What's it?

Reply:
If the database has a Full Text Index, the function is called with “<scope>.<collection>::<index name>. This violates one of the checks, and therefore should return { nullslice, nullslice } as it is not a collection. In the test I added the REQURE(c4coll_deleteIndex) to ensure the index is still available once the database has been re-opened, and it is. So the change I’ve made here produces what the expected behaviour should have been.

Reply:
Github seems to have taken some of my formatting, function is called with [scope].[collection]::[index name]

Reply:
I see. We probably should handle this exception rather than taking away the assertion altogether. Also make sure whether we should return the collection part of the key store name of simply null based on the calling code thread.

Reply:
Per the comments above it, that function’s purpose is just to determine if whatever’s passed to it is a collection. And it’s up to the caller whether { nullptr, nullptr } is expected return or not. So for a FTI, it is expected to get back { nullptr, nullptr } as it is not a collection. Additionally, as it is DebugAssert, it will not do anything when built for release (the assertion only runs when building in Debug mode).

Reply:
Key table names are completely derived by us (the program). We should know the format. Here we encountered a surprise. We can account it. I don't at the moment whether we should return the collection part or null, but at least we should note that this a valid case.

Reply:
I’m not sure, I could debug it more and see where it’s called from. But due to the description above the function: “ If the KeyStore is not a collection, the collection & scope names will be `nullslice`”, I assumed it is expected that not all KeyStores passed to the function will be collections. 
My assumption is that all keyStores are passed to this function upon loading the DB to determine which are collections.

Reply:
Although thinking about it, this issue is only encountered with Full Text Indexes, so a question I have is: Are all indexes saved as a keystore? If so, it may be that for some reason, other indexes are not being passed to this function, but FTIs are.

Reply:
Usually indexes are simply created in SQLite, and then recorded in an indexes table to ensure that they are not created again.  FTS indexes are special though because the full text engine inside of SQLite requires the use of virtual tables in order to function.